### DISCUSSION TITLE
Macro facility

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#320] [Ideas]
I quite often thought that a CREXX macro facility would be helpful:
1. especially in levelb there are quite often combinations of several instructions used, 
2. in some situations you can use a macro call instead of inserting a new instruction, 
3. in general (Level c) a macro call (code is inserted) is cheaper than calling a function
4. I love languages with a Macro facility

Ideally, a macro could be a REXX itself inserting the generated lines into the code.  
I can't judge how difficult it is to implement an additional pass into RXC.

**Comment by adesutherland** [comment]
I note the

    I love languages with a Macro facility
:-)

... but can you see what the inlining and calling procedures and variables between files is finally like and see if there is still a need? 

Macros can become ugly ... but if there is a need we can probably write it in crexx levelb which would be nice
